enclave/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+add_library(oecryptombed STATIC
+    ../../common/asn1.c
+    ../../common/cert.c
+    ../../common/kdf.c
```

Comment:
Since kdf.c does not have any mbedtls (or openssl) specific calls, does it need to be here? All the other .c files that you are including here require calls unique to the specific crypto that are used.

Reply:
@CodeMonkeyLeet @mikbras We need a discussion on what is part of the OE crypto API. The API was originally made as a way to share code with a common API for the host and the enclave, but now we're using the API for pluggable crypto, which is a different use case.

Reply:
So, all the code in common (`asn1.c`, `cert.c`, `kdf.c`) can be purely through the oecrypto APIs. However, if you take these files and move them out to `oeenclave`, you run into linking problems, since the enclave specific code in `oecryptombed` might refer to a symbol that's defined in `common/[asn1.c|cert.c|kdf.c]`.

Speficially, the issue I hit was that `common/asn1.c` defines a functions that is used by `enclave/crypto/asn1.c` so the linking fails.

Reply:
We can merge this pull request and continue refactoring incrementally from then on.

Reply:
Yeah, a similar discussion just came up in #1825. When there is further incremental work to be done, we should capture that in an issue(s) for the current release's milestone, and allow the partial implementation to be merged in order to avoid long-lived branches.